### DISCUSSION TITLE
Xarray Accessor for Georeferenced Data Comparisons

### DISCUSSION THREAD

**Discussion by fernando-aristizabal** [pydata/xarray#7649] [General]
### What is your issue?

Dear Xarray Community,

We are exploring the possibility of extending xarray with a new accessor and would like to share our idea with you as recommended on the [extending xarray page](https://docs.xarray.dev/en/stable/internals/extending-xarray.html).

Our proposal involves creating an xarray accessor that compares georeferenced DataArrays and Datasets, as per [rioxarray](https://github.com/corteva/rioxarray). The comparisons will be based on scoring philosophies for three statistical data types: categorical, continuous, and probabilistic. The accessor functions will need to process and align the xarrays before performing comparisons. Once the georeferenced xarrays are homogenized, an agreement xarray can be computed, with its structure varying based on the statistical data type used. The comparison process will also generate agreement metrics, which will also depend on the data types involved. Furthermore, we aim to incorporate attributes from the compared datasets into the resulting agreement outputs. These attributes might be sourced directly from the data files or potentially integrated through a cataloging approach.

We acknowledge the existence of established projects like [climpred](https://github.com/pangeo-data/climpred) along with [xskillscore](https://github.com/xarray-contrib/xskillscore), but we feel that at least climpred might be too domain-specific to climate data. As our goal is to work with 2D/3D raster data models found within GDAL, we believe the development of this package is warranted.

We appreciate your time in reviewing our proposal and welcome any feedback or suggestions you may have.

Thank you for your consideration.

**Comment by dcherian** [comment]
Thanks @fernando-aristizabal It may be better to discuss at discourse.pangeo.io and/or https://github.com/corteva/rioxarray/discussions That community tends to hang out there more IMO.

**Reply by fernando-aristizabal** [reply]
Thank you @dcherian.
